generate.py

```diff
@@ -137,10 +136,9 @@ def main(
             temperature=temperature,
             top_k=top_k,
         )
+        t = time.perf_counter() - t0
         print(tokenizer.decode(y))
-
-    t = time.perf_counter() - t0
-    print(f"\n\nTime for inference: {t:.02f} sec total, {num_samples * max_new_tokens / t:.02f} tokens/sec", file=sys.stderr)
+        print(f"Time for inference {i + 1}: {t:.02f} sec total, {max_new_tokens / t:.02f} tokens/sec", file=sys.stderr)
```

Comment:
What if we just drop the total time and print the tok/sec for the last iteration of the loop (unindenting the print)? The total time is not that useful tbh. The averaging over just a single generate() call should be good enough as estimate for tok/sec.

Also, the changes would have to be made for the other generate_* scripts.

Reply:
I personally like having the extra information for each generation. Anyways most people will run `num_samples=1` so there's no difference in that case.